synapse/http/federation/srv_resolver.py

```diff
@@ -135,7 +136,7 @@ async def resolve_service(self, service_name: bytes) -> List[Server]:
 
         try:
             answers, _, _ = await make_deferred_yieldable(
-                self._dns_client.lookupService(service_name)
+                self._dns_client.lookupService(service_name, timeout=(1, 3, 11, 35))
```

Comment:
1 + 3 + 11 + 35 = 50
---
By the way, i saw somewhere another federation-esc timeout of 30 seconds, shouldn't this timeout sum become 25 seconds? Or is 50 seconds acceptable?

Reply:
so what's going on here? we're trying to change the timeout so that it's less than the total federation request timeout, so that we get a useful error message?

if so, it would be useful to say that in a comment.

(looks like the defaults are `(1, 3, 11, 45)`. could you include that in the comment too)